### PR TITLE
[S3 Repository] Make bulk delete size configurable

### LINKED ISSUE #10451: [S3 repository plugin] Configurable bulk delete size
Currently bulk size is hardcoded to 1000 . For remote store , this leads to a spike in deletion calls. Making it configurable will help the call rate to s3 smoothen out .

### PR DESCRIPTION
**Pull request by gbbafna**
### Description
Make bulk delete size configurable

### Related Issues
Resolves #10451

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Repository.java [hunk 1]
```diff
@@ -190,6 +190,13 @@ class S3Repository extends MeteredBlobStoreRepository {
         new ByteSizeValue(5, ByteSizeUnit.TB)
     );
 
+    /**
+     * Maximum number of deletes in a DeleteObjectsRequest.
+     *
+     * @see <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/multiobjectdeleteapi.html">S3 Documentation</a>.
+     */
+    static final Setting<Integer> BULK_DELETE_SIZE = Setting.intSetting("bulk_delete_size", 1000, 1, 1000);
```
Comment (ashking94) [inline plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Repository.java:198]: Lets have minimum as 0 to allow deletes to not happen if required by the user?

**Review by ashking94 — APPROVED** [review]
lgtm

File: plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Repository.java (hunk 1 above)
Comment (gbbafna) [reply plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Repository.java:198]: 0 is a breach of contract for the API, hence not going ahead with that.

**Review by sachinpkale — APPROVED** [review]

File: plugins/repository-s3/src/test/java/org/opensearch/repositories/s3/S3BlobContainerRetriesTests.java [hunk 2]
```diff
@@ -215,6 +216,7 @@ protected AsyncMultiStreamBlobContainer createBlobContainer(
                 bufferSize == null ? S3Repository.BUFFER_SIZE_SETTING.getDefault(Settings.EMPTY) : bufferSize,
                 S3Repository.CANNED_ACL_SETTING.getDefault(Settings.EMPTY),
                 S3Repository.STORAGE_CLASS_SETTING.getDefault(Settings.EMPTY),
+                BULK_DELETE_SIZE.get(Settings.EMPTY),
```
[the remark this answers is not in the document]
Comment (gbbafna) [reply plugins/repository-s3/src/test/java/org/opensearch/repositories/s3/S3BlobContainerRetriesTests.java:219]: added it now.

**Comment by gbbafna** [comment]
```

* What went wrong:
Execution failed for task ':modules:lang-expression:internalClusterTest'.
``` 

Gradle check failure is not related . Will retry .

**Pull request merged**